parser.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+// Parser Structure: dictionary, sections
+// It's basically a map of a map, map of sections where each section maps to keys : values
+// map[section] --> returns a map where each key --> value
+type Parser struct {
+	dictionary map[string]map[string]string
+}
+
+// NewParser creates an instance of our Parser, which is iniitially an empty map of maps
+func NewParser() Parser {
+	return Parser{make(map[string]map[string]string)}
+}
+
+// ErrCouldNotOpen happens when file cannot be opened and provides file name
+var ErrCouldNotOpen error
+
+// ErrMissingValueAssignment happens when a key isn't followed by an = statement
+var ErrMissingValueAssignment error
+
+// ErrSectionNameMissingClosure happens when section name is missing the ] paranthesis
+var ErrSectionNameMissingClosure error
+
+// ErrWrongParanthesisOrder happens when section name starts by the wrong paranthesis ']'
+var ErrWrongParanthesisOrder error
+
+// ErrInvalidSectionName happens when section is written in a wrong form --> ex: sectionName]
+var ErrInvalidSectionName error
+
+// ErrCouldNotWriteToFile happens when file cannot be written to
+var ErrCouldNotWriteToFile error
```

Comment:
> why empty values

Because these are not static errors, at first I initialized them here but eng/thabet said it would be better to provide users wit info as we can, that's why initialization is made in parsing function, in order to pass the right values (dynamic errors)

Reply:
so you can just define/return errors at once when needed no?

Reply:
> so you can just define/return errors at once when needed no?

yes we can, however I used their names in testing to ensure that I trach errors correctly in the test file, function:
`func Test_LoadFromFile(t *testing.T)` t